packages/e2e-tests/specs/editor/various/links.test.js

```diff
@@ -26,6 +26,42 @@ describe( 'Links', () => {
 		);
 	};
 
+	it( 'will use Post title as link text if link to existing post is created without any text selected', async () => {
+		const titleText = 'Post to create a link to';
+		await createPostWithTitle( titleText );
+
+		await createNewPost();
+		await clickBlockAppender();
+
+		// Now in a new post and try to create a link from an autocomplete suggestion using the keyboard.
+		await page.keyboard.type( 'Here comes a link: ' );
+
+		// Press Cmd+K to insert a link
+		await pressKeyWithModifier( 'primary', 'K' );
+
+		// Wait for the URL field to auto-focus
+		await waitForAutoFocus();
+		expect(
+			await page.$(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).not.toBeNull();
+
+		// Trigger the autocomplete suggestion list and select the first suggestion.
+		await page.keyboard.type( titleText.substr( 0, titleText.length - 2 ) );
+		await page.waitForSelector( '.block-editor-link-control__search-item' );
+		await page.keyboard.press( 'ArrowDown' );
+
+		await page.keyboard.press( 'Enter' );
+
+		const actualText = await page.evaluate(
+			() =>
+				document.querySelector( '.block-editor-rich-text__editable a' )
+					.textContent
```

Comment:
It's only a small point and probably won't make much of a difference at least in the short-term, but this is a pretty broad selector. While it might not seem like it, there's often many instances of block elements and/or rich text elements in a given page, so the fact that we hope the first one we match is the one we expect it to be could potentially bite us in the future if those circumstances were to change.

Generally, it is quite hard to pinpoint the specific block we're looking for. 

A few approaches I've seen and used in the past:

- A selector which identifies blocks within the block list, then picks the one at the expected index
- Selecting from where we expect focus to be, if focus is assumed to return to the field after the link is applied
   - Something like `await page.evaluate( () => document.activeElement.closest( '.block-editor-rich-text__editable a' ) );`

Reply:
I had a pretty hard time figuring out the test, so I was happy when I finally saw mine succeed 😄. I'll keep your suggestions in mind for any future tests